src/zst_decompressor.ts

```diff
@@ -0,0 +1,237 @@
+// const SIZE_LIMIT = 30 * 1024 * 1024; // 30MB
+const SIZE_LIMIT = 30 * 1024 * 1024; // 0.5MB, for testing
+
+// set the total size limit of the combined decompressed files (this is how much space you need to have available on your PC prior to running node src/streaming_partial_decompresser.js)
+const decompressedSizeLimit = 500 * 1024 * 1024 * 1024; // 500 GB represented in bytes
+
+const getFileSize = (filePath: string) => {
+  if (!fs.existsSync(filePath)) {
+    return 0;
+  }
+  const stats = fs.statSync(filePath);
+  return stats.size;
+};
+
+/**
+ * Runs the analysis script on a given file path.
+ * @param {string} filePath - The path of the file to run the analysis on.
+ * @return {Promise} A promise that resolves when the analysis is complete.
+ */
+async function runAnalysis(filePath: string) {
+  return new Promise<void>((resolve, reject) => {
+    // Run the analysis script
+    console.log(`Running analysis script on ${filePath}...`);
+
+    const child = spawn('ts-node', [
+      //   '/Users/bennyrubanov/Coding_Projects/chessanalysis/src/index_with_decompressor.ts',
+      `${__dirname}/../../run_metrics_on_input.ts`,
+      filePath,
+    ]);
+
+    // only log complete lines of output (no insertion of "stdout: " in the middle of a line)
+    // do this by accumulating the data until a newline character, and then logging the accumulated data
+    let accumulatedData = '';
+    child.stdout.on('data', (data) => {
+      accumulatedData += data;
+
+      let newlineIndex;
+
+      // this loop slices data while theere is a newline chanracter in the accumulated data
+      while ((newlineIndex = accumulatedData.indexOf('\n')) >= 0) {
+        console.log(`stdout: ${accumulatedData.slice(0, newlineIndex)}`);
+        accumulatedData = accumulatedData.slice(newlineIndex + 1);
+      }
+    });
+
+    child.stderr.on('data', (data) => {
+      console.log(`stderr: ${data}`);
+    });
+
+    child.on('error', (error) => {
+      console.log(`error: ${error.message}`);
+      reject(error);
+    });
+
+    child.on('close', (code) => {
+      console.log(`child process exited with code ${code}`);
+      resolve();
+    });
+  });
+}
+
+/**
+ * Decompresses and analyzes a file.
+ *
+ * @param {string} file - The name of the file to decompress and analyze.
+ * @param {number} [start=0] - The starting index for decompression.
+ * @returns {Promise} A promise that resolves when the decompression and analysis is complete.
+ */
+const decompressAndAnalyze = async (file, start = 0) => {
+  let these_chunks_counter = 0; // Initialize the chunk counter
+  let file_counter = 1; // Initialize the file counter
+  let total_chunk_counter = 0;
+  const filesProduced = new Set();
+
+  //   const base_path = `/Users/bennyrubanov/Coding_Projects/chessanalysis/data/${file.replace(
+  const base_path = `${__dirname}/../../data/${file.replace('.zst', '')}`;
+
+  // Create a new file path
+  const newFilePath = `${base_path}_${randomUUID()}`;
```

Comment:
does randomUUID create a random file path number? i prefer counting them. corresponds better with the log here too  `console.log(`Creating file #${file_counter} at ${newFilePath}`);`

Reply:
the issue with counting is that in high concurrency they will not match up. We could do it by game identifier or institute locking if necessary, but for uniquness I use UUID

Reply:
let's do both? one for logs, one for avoiding concurrency/overwrites?